diff --git a/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/api/TezRuntimeConfiguration.java b/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/api/TezRuntimeConfiguration.java
index 08f76f2..462b257 100644
--- a/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/api/TezRuntimeConfiguration.java
+++ b/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/api/TezRuntimeConfiguration.java
@@ -524,6 +524,11 @@ public class TezRuntimeConfiguration {
       "merge.progress.records";
   public static final long TEZ_RUNTIME_RECORDS_BEFORE_PROGRESS_DEFAULT = 10000;
 
+  @Unstable
+  @ConfigurationProperty
+  public static final String TEZ_RUNTIME_GLOBAL_SORTED_TABLE =
+  TEZ_RUNTIME_PREFIX + "global.sorted.table";
+  
   static {
     tezRuntimeKeys.add(TEZ_RUNTIME_IFILE_READAHEAD);
     tezRuntimeKeys.add(TEZ_RUNTIME_IFILE_READAHEAD_BYTES);
@@ -585,6 +590,7 @@ public class TezRuntimeConfiguration {
     tezRuntimeKeys.add(TEZ_RUNTIME_CONVERT_USER_PAYLOAD_TO_HISTORY_TEXT);
     tezRuntimeKeys.add(TEZ_RUNTIME_SORTER_CLASS);
     tezRuntimeKeys.add(TEZ_RUNTIME_CLEANUP_FILES_ON_INTERRUPT);
+    tezRuntimeKeys.add(TEZ_RUNTIME_GLOBAL_SORTED_TABLE);
 
     defaultConf.addResource("core-default.xml");
     defaultConf.addResource("core-site.xml");
diff --git a/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/output/OrderedPartitionedKVOutput.java b/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/output/OrderedPartitionedKVOutput.java
index c0b0760..f95adb5 100644
--- a/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/output/OrderedPartitionedKVOutput.java
+++ b/tez-runtime-library/src/main/java/org/apache/tez/runtime/library/output/OrderedPartitionedKVOutput.java
@@ -245,6 +245,8 @@ public class OrderedPartitionedKVOutput extends AbstractLogicalOutput {
     confKeys.add(TezConfiguration.TEZ_COUNTERS_MAX_GROUPS);
     confKeys.add(TezRuntimeConfiguration.TEZ_RUNTIME_SORTER_CLASS);
     confKeys.add(TezRuntimeConfiguration.TEZ_RUNTIME_CLEANUP_FILES_ON_INTERRUPT);
+    confKeys.add(TezRuntimeConfiguration.TEZ_RUNTIME_GROUP_COMPARATOR_CLASS);
+    confKeys.add(TezRuntimeConfiguration.TEZ_RUNTIME_GLOBAL_SORTED_TABLE);
   }
 
   // TODO Maybe add helper methods to extract keys
diff --git a/tez-ui/src/main/webapp/node_tmp/node-v6.11.3-linux-x64/bin/npm b/tez-ui/src/main/webapp/node_tmp/node-v6.11.3-linux-x64/bin/npm
new file mode 100755
index 0000000..e69de29
diff --git a/tez-ui/src/main/webapp/node_tmp/node.tar.gz b/tez-ui/src/main/webapp/node_tmp/node.tar.gz
new file mode 100644
index 0000000..d789811
Binary files /dev/null and b/tez-ui/src/main/webapp/node_tmp/node.tar.gz differ
